thinc/util.py

```diff
@@ -203,12 +203,20 @@ def copy_array(dst: ArrayXd, src: ArrayXd) -> None:  # pragma: no cover
         numpy.copyto(dst, src)  # type: ignore
 
 
-def to_categorical(Y: IntsXd, n_classes: Optional[int] = None) -> FloatsXd:
+def to_categorical(
+    Y: IntsXd,
+    n_classes: Optional[int] = None,
+    label_smoothing: float = 0.0,
+) -> FloatsXd:
+    assert label_smoothing >= 0 and label_smoothing < 0.5
```

Comment:
0.5 seems like a very high value for label smoothing? Is that realistic?

Reply:
Also, this shouldn't be an `assert`, but a proper value check + raising an informative error, because the value can be a mistake by a user of Thinc. `assert` should only be used to check internal consistencies, not for catching wrong input.

Reply:
I agree that it's the appropriate mathematical upperbound, it just feels like way too drastic.

Actually, just checked over at TF and they use the interval [0, 1] there. That's just ... silly. Anyway I guess 0.5 is fine, though I couldn't imagine anything above 0.2 being really useful in actual applications.